iiod/CMakeLists.txt

```diff
@@ -90,8 +90,22 @@ if(NOT SKIP_INSTALL_ALL)
 endif()
 
 if (WITH_SYSTEMD)
+	if(${WITH_HWMON})
+		set(CMAKE_SERVICE_PATH "ConditionPathExists=|/sys/bus/iio\nConditionPathExists=|/sys/class/hwmon")
+	else()
+		set(CMAKE_SERVICE_PATH "ConditionPathExists=/sys/bus/iio")
+	endif()
 	configure_file(${CMAKE_CURRENT_SOURCE_DIR}/init/iiod.service.cmakein ${PROJECT_BINARY_DIR}/init/iiod.service)
 	install(FILES ${PROJECT_BINARY_DIR}/init/iiod.service DESTINATION ${SYSTEMD_UNIT_INSTALL_DIR})
+	find_program(SYSTEMCTL
+		NAMES systemctl
+		DOC "Control the systemd system and service manager (systemctl)")
+	if(SYSTEMCTL)
```

Comment:
The part inside this if() will fail when cross-compiling

Reply:
Yeah, moreover the change is not related with the current patch. And I honestly don't think handling like the below belongs to the build system. We should either let it for users or if we really want for the service to be automatically enabled, do it at the packet manager level.